src/backend/looped/session/auto_kick_host.cpp

```diff
@@ -12,6 +12,10 @@ namespace big
 		if (kick_host && !bLastKickHost)
 		{
 			g_player_service->iterate([](auto& plyr) {
+				// Don't kick trusted players
+				if ((plyr.second->is_friend() && g.session.trust_friends) || plyr.second->is_trusted)
```

Comment:
Order of operations:

1. If they are trusted.
2. If we have trust_friends on
3. if they're a friend

This will give optimization notes to the compiler. We don't care if they're a friend if we have trust_friends disabled. We also don't care about either of those if they're already trusted.